src/ks_json.c

```diff
@@ -451,6 +451,64 @@ KS_DECLARE(ks_json_t *) ks_json_enum_next(ks_json_t *item)
 	return item->next;
 }
 
+KS_DECLARE(ks_bool_t) ks_json_replace_item(ks_json_t *parent, ks_json_t **item, ks_json_t *replacement)
+{
+	if ((parent == NULL) || (replacement == NULL) || (item == NULL) || (*item == NULL)) {
+		return KS_FALSE;
+	}
+
+	if (replacement == *item) {
+		*item = NULL;
+
+		return KS_TRUE;
+	}
+
+	replacement->next = (*item)->next;
+	replacement->prev = (*item)->prev;
+
+	if ((*item)->string) {
+		if (replacement->string) {
+			free(replacement->string);
+		}
+
+		replacement->string = strdup((*item)->string);
+	}
+
+	if (replacement->next != NULL) {
+		replacement->next->prev = replacement;
+	}
+
+	if (replacement->prev != NULL) {
+		replacement->prev->next = replacement;
+	}
+
+	if (parent->child == *item) {
+		parent->child = replacement;
+	}
+
+	(*item)->next = NULL;
+	(*item)->prev = NULL;
+
+	ks_json_delete(item);
+
+	return KS_TRUE;
+}
+
+KS_DECLARE(ks_bool_t) ks_json_replace_value_string(ks_json_t *string_item, char *value_string)
+{
+	if (string_item == NULL || !ks_json_type_is_string(string_item)) {
+		return KS_FALSE;
+	}
+
+	if (string_item->valuestring != NULL) {
+		free(string_item->valuestring);
```

Comment:
imho `free` is not safe. There may be a special hook. See https://github.com/signalwire/libks/blob/master/src/include/cJSON/cJSON.c#L170-L184

```
CJSON_PUBLIC(void) kJSON_InitHooks(kJSON_Hooks* hooks)
{
    if (hooks == NULL)
    {
        /* Reset hooks */
        global_hooks.allocate = malloc;
        global_hooks.deallocate = free;
        global_hooks.reallocate = realloc;
        return;
    }

	global_hooks.allocate = hooks->malloc_fn;
	global_hooks.deallocate = hooks->free_fn;
	global_hooks.reallocate = hooks->realloc_fn;
}
```

Reply:
Sounds good to me
```
CJSON_PUBLIC(void) kJSON_free(void *object)
{
    global_hooks.deallocate(object);
}
```